File: C#/Funcionarios/Funcionario.cs
using System;
using ByteBank.SistemaInterno;

namespace ByteBank.Funcionarios
{
    public abstract class Funcionario : Conta
    {
        // Atributos (Getters e Setters)
        public string Profissao { get; set; }
        public double Salario { get; set; }

        // Construtor
        protected Funcionario(string nome, double cpf, double saldo) : base(nome, cpf, saldo) { }


        // Métodos
        public override void Extrato()
        {
            Console.WriteLine("Nome: " + Nome);
            Console.WriteLine("CPF: " + Cpf);
            Console.WriteLine("Saldo: " + Saldo);
            Console.WriteLine("Salário: " + Salario);
        }

        public abstract void ReceberSalario();
        public abstract void ReceberAumento();
    }
}

File: Java/src/Funcionarios/Funcionario.java
package Funcionarios;

import Sistema.Conta;

public abstract class Funcionario extends Conta {

    // Atributos
    private double salario;
    private boolean demissao;

    // Construtor
    public Funcionario(String nome, double cpf, double saldo, double numeroConta) {
        super(nome, cpf, saldo, numeroConta);
    }


    // Métodos
    public void pedirDemissao() {
        this.demissao = true;
        System.out.println(this.getNome() + ", foi demitido.");
    }

    // Métodos Abstratos
    public abstract void recebeSalario();
    public abstract void recebeBonificacao();


    // Getters e Setters
    public double getSalario() {
        return salario;
    }

    public void setSalario(double salario) {
        this.salario = salario;
    }

    public boolean isDemissao() {
        return demissao;
    }

    public void setDemissao(boolean demissao) {
        this.demissao = demissao;
    }
}
